src/Documentation/grains/transactions.md

```diff
@@ -88,31 +116,54 @@ public class AccountGrain : Grain, IAccountGrain
     private readonly ITransactionalState<Balance> balance;
 
     public AccountGrain(
-        [TransactionalState("balance")] ITransactionalState<Balance> balance)
+        [TransactionalState("balance", "TransactionStore")]
+        ITransactionalState<Balance> balance)
     {
         this.balance = balance ?? throw new ArgumentNullException(nameof(balance));
     }
 
     Task IAccountGrain.Deposit(uint ammount)
     {
-        this.balance.State.Value += ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value += amount);
     }
 
     Task IAccountGrain.Withdrawal(uint ammount)
     {
-        this.balance.State.Value -= ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value -= amount);
```

Comment:
I know this is an example, but there should be logic here to check for integer underflow. Otherwise, subtracting 1 from a balance of 0 makes the user a billionaire and we are relying on logic specified in `ATMGrain`, which I cannot see defined here.

Reply:
This is beyond my current skill level. Can you send an example, please?

Also, I removed the extra 'm' from amount. That IS within my skill level.

Reply:
This is just meant to show the pattern.  We can improve the logic but I don't think that will make the doc any clearer.  same with changing uint to int.